File: 896 - Monotonic Array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& A) {
        bool res = true, inc = false, dec = false;
        for (int i = 0; i < A.size() - 1; i++) {
            if (A[i] < A[i + 1]) inc = true;
            else if (A[i] > A[i + 1]) dec = true;
            if (inc && dec) {
                res = false;
                break;
            }
        }
        return res;
    }
};

File: 896 - Monotonic Array.py
class Solution:
    def isMonotonic(self, A: list[int]) -> bool:
        res = True
        inc = False
        dec = False
        for i in range(len(A) - 1):
            if A[i] > A[i + 1]:
                dec = True
            elif A[i] < A[i + 1]:
                inc = True
            if inc and dec:
                res = False
                break
        return res